src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs

```diff
@@ -0,0 +1,218 @@
+        {
+            await PostHelper(serverUri, ExpectedContent, new RepeatedFlushContent(ExpectedContent), false, true);
+        }
+
+        [Theory, MemberData("EchoServers")]
+        public async Task PostUsingUsingConflictingSemantics_UsesChunkedSemantics(Uri serverUri)
+        {
+            await PostHelper(serverUri, ExpectedContent, new StringContent(ExpectedContent), true, true);
+        }
+
+        [Theory, MemberData("EchoServers")]
+        public async Task PostUsingNoSpecifiedSemantics_UsesChunkedSemantics(Uri serverUri)
+        {
+            await PostHelper(serverUri, ExpectedContent, new StringContent(ExpectedContent), false, false);
+        }
+
+        [Theory, MemberData("BasicAuthEchoServers")]
+        public async Task PostRewindableContentUsingAuth_NoPreAuthenticate_Success(Uri serverUri)
+        {
+            HttpContent content = CustomContent.Create(ExpectedContent, true);
+            var credential = new NetworkCredential(UserName, Password);
+            await PostUsingAuthHelper(serverUri, ExpectedContent, content, credential, false);
+        }
+
+        [Theory, MemberData("BasicAuthEchoServers")]
+        public async Task PostNonRewindableContentUsingAuth_NoPreAuthenticate_ThrowsInvalidOperationException(Uri serverUri)
+        {
+            HttpContent content = CustomContent.Create(ExpectedContent, false);
+            var credential = new NetworkCredential(UserName, Password);
+            await Assert.ThrowsAsync<InvalidOperationException>(() => 
+                PostUsingAuthHelper(serverUri, ExpectedContent, content, credential, false));
+        }
+
+        [Theory, MemberData("BasicAuthEchoServers")]
+        public async Task PostNonRewindableContentUsingAuth_PreAuthenticate_Success(Uri serverUri)
+        {
+            HttpContent content = CustomContent.Create(ExpectedContent, false);
+            var credential = new NetworkCredential(UserName, Password);
+            await PostUsingAuthHelper(serverUri, ExpectedContent, content, credential, true);
+        }
+        
+        private async Task PostHelper(
+            Uri serverUri,
+            string requestBody,
+            HttpContent requestContent,
+            bool useContentLengthUpload,
+            bool useChunkedEncodingUpload)
+        {
+            using (var client = new HttpClient())
+            {
+                if (!useContentLengthUpload)
+                {
+                    requestContent.Headers.ContentLength = null;
+                }
+                
+                if (useChunkedEncodingUpload)
+                {
+                    client.DefaultRequestHeaders.TransferEncodingChunked = true;
+                }
+
+                HttpResponseMessage response = await client.PostAsync(serverUri, requestContent);
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+                var responseContent = await response.Content.ReadAsStringAsync();
```

Comment:
@stephentoub 

>Does the response message need to be disposed, or does ReadAsStringAsync take care of any such necessary cleanup?

`ReadAsStringAsync` does NOT dispose of the `HttpResponseMessage`.  It's not critical to Dispose of it in terms of this particular test.  But it is a good practice to Dispose of it when done, unless we are specifically testing Dispose scenarios.

I'll add the Dispose here for best practice.